### PR TITLE
Nikkxll/profile new features

### PR DESCRIPTION
**Pull request by nikkxll**
- Added new features to the profile (pass + username change, some stats)
- Added winRate and totalMatches params to the tournament and single game lobbies
- Minor changes and fixes

### CODE REVIEW

File: frontend/src/js/profileSettings.js [hunk 1]
```diff
@@ -3,250 +3,268 @@
...
+    const friends = await friendsRequest.json();
+
+    const json = await response.json();
+    if (json.data.players.length == 0) matches.innerText = "No one else exists";
+    else {
+      let requestPlayers = "";
+      let friendPlayers = "";
+      let otherPlayers = "";
+      const playersArr = json.data.players.filter(
+        (player) => player.username != "ai_player"
```
Comment (madfcat) [inline frontend/src/js/profileSettings.js]:
```suggestion
        (player) => (player.username !== "ai_player" && player.id !== userId)
```

**Review by madfcat — CHANGES_REQUESTED** [review]

File: frontend/src/js/profileSettings.js (hunk 1 above)
Comment (nikkxll) [reply frontend/src/js/profileSettings.js]: Changed
Thx!

**Comment by madfcat** [comment]
Updating `username` when user has no `displayName`, does not update it in the lobby and in the recent matches.

Also, consider adding username to the profile, so a user can see the initial username. And update it after changes.

No consistency between the button hovers. I guess it can be added later if needed.

**Comment by nikkxll** [comment]
Added features and fixes in accordance with the @madfcat last comment

**Comment by madfcat** [comment]
`username` input field should be emptied after successful update.
At the moment, it still has the last value after update.

**Comment by madfcat** [comment]
Please, update styles like this (styles.css line 159):
```
input[type="file"] {
	color: transparent;
	width: 100%;
  }
  
input[type="file"]::file-selector-button {
	display: block;
	margin: 0 auto;
	border-radius: 5px;
	border: unset;
	cursor: pointer;
	transition: background-color 0.3s;
}
input[type="file"]::file-selector-button:hover {
	background-color: #000000;
	color: rgba(255, 255, 255, 0.5);
}
```

It will make the "Choose file" button have the same style.

**Comment by nikkxll** [comment]
Updated

**Review by madfcat — APPROVED** [review]
Looking good!

**Pull request merged**